client/auth/AuthProvider.tsx

```diff
@@ -37,12 +28,30 @@ const CHECK_AUTH_QUERY = gql`
     }
 `;
 
-function AuthProvider({ children }: MyComponentProps) {
-    const { data } = useQuery(CHECK_AUTH_QUERY);
-    if (data && data.checkAuth !== null) {
-        setAuthContext(data.checkAuth);
-    }
-    return <AuthContext.Provider value={authContext}>{children}</AuthContext.Provider>;
+function AuthProvider({ children }: React.PropsWithChildren<unknown>) {
+    const [user, setUser] = useState<UserContext | null>(null);
+    const [loading, setLoading] = useState(true);
+
+    // query the server to check if there is a valid session
+    useEffect(() => {
+        const { data } = useQuery(CHECK_AUTH_QUERY);
+        if (data && data.checkAuth !== null) {
+            setLoading(false);
+            setUser({
+                email: data.checkAuth.email,
+                id: data.checkAuth.id,
+            });
+        }
+    }, []);
+    const value: AuthContextType = useMemo(
```

Comment:
What benefit do we get with `useMemo` here?

Reply:
eslint was throwing an error and recommend implementing it. I have since removed it at escaped the error.